lib/nosedrum/application_command.ex

```diff
@@ -95,10 +95,21 @@ defmodule Nosedrum.ApplicationCommand do
 
   alias Nostrum.Struct.{Embed, Interaction}
 
+  @typedoc """
+  A function called by `Nosedrum.Storage.followup/2` after deferring an interaction response.
+  """
+  @type callback :: {fun(), args :: list()} | {module(), fn_name :: atom(), args :: list()}
```

Comment:
The second one is already built in 🙂 
```suggestion
  @type callback :: {fun(), args :: list()} | mfa()
```

See https://hexdocs.pm/elixir/1.12/typespecs.html

Reply:
Ooo I didn't know about `mfa()`, though is `arity()` correct for the third tuple element? It's described as a range `0..255`, but I would expect it to be a list of args

Reply:
Committed that for now since it's still clear enough. I also expanded the function spec in the F/A variant to specify it returns a `response()` as you suggested, though I'm not sure of a way to do that with an MFA tuple unfortunately.

Hopefully José & friends will bring us set-theoretic types one day :')

Reply:
ahhh - no, you are right of course, it should be a list of args instead of an MFA, so `{module(), fn_name :: atom(), args :: list()}` was the correct thing to use!

Reply:
Okay that makes sense, reverted!